From ac93e92992f5a6b17883367840148ea62d855434 Mon Sep 17 00:00:00 2001
From: Toby Flynn <tflynn@redwiretechnology.com>
Date: Thu, 30 Nov 2023 15:05:29 -0500
Subject: [PATCH] pass through fpga type information

---
 board/xilinx/zynqmp/zynqmp.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/board/xilinx/zynqmp/zynqmp.c b/board/xilinx/zynqmp/zynqmp.c
index 579708d2e0..029fcad71a 100644
--- a/board/xilinx/zynqmp/zynqmp.c
+++ b/board/xilinx/zynqmp/zynqmp.c
@@ -522,6 +522,13 @@ int board_late_init(void)
 			env_targets ? env_targets : "");
 
 	env_set("boot_targets", new_targets);
+    
+    #if defined(CONFIG_FPGA) && defined(CONFIG_FPGA_ZYNQMPPL) && \
+            !defined(CONFIG_SPL_BUILD) || (defined(CONFIG_SPL_FPGA_SUPPORT) && \
+            defined(CONFIG_SPL_BUILD))
+            env_set("fpgadev",  &zynqmppl.name[2]);
+    #endif
+    
 	free(new_targets);
 
 	reset_reason();
